plaid/src/apis/npm/npm_web_client.rs

```diff
@@ -0,0 +1,833 @@
+        // This will later be sent as form data when actually performing the deletion.
+        let response = self
+            .client
+            .get(format!(
+                "{}/package/%40{}%2F{}/delete",
+                NPMJS_COM_URL, self.config.npm_scope, package
+            ))
+            .send()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToDeletePackage))?;
+        // Safe unwrap: hardcoded data.
+        // TODO best to precompile regex? This is likely an infrequent operation.
+        let dsr_manifest_hash = Regex::new(r#"dsrManifestHash\"\s+?value=\"([a-z0-9]{64})\""#)
+            .unwrap()
+            .captures(
+                &response
+                    .text()
+                    .await
+                    .map_err(|_| ApiError::NpmError(NpmError::FailedToDeletePackage))?,
+            )
+            .ok_or(ApiError::NpmError(NpmError::FailedToDeletePackage))?
+            .get(1) // get the content of the capturing group, which contains the token we need
+            .ok_or(ApiError::NpmError(NpmError::FailedToDeletePackage))?
+            .as_str()
+            .to_string();
+
+        // Step 2. Perform the actual package deletion with a POST request to /delete
+        let scoped_package_name = format!("@{}/{}", self.config.npm_scope, package);
+        let csrf_token = self
+            .get_csrftoken_from_cookies()
+            .map_err(|_| ApiError::NpmError(NpmError::WrongClientStatus))?;
+        let form_data: HashMap<&str, &str> = [
+            ("package", scoped_package_name.as_str()),
+            ("dsrManifestHash", &dsr_manifest_hash),
+            ("csrftoken", &csrf_token),
+        ]
+        .into();
+        self.client
+            .post(format!(
+                "{}/package/%40{}%2F{}/delete",
+                NPMJS_COM_URL, self.config.npm_scope, package
+            ))
+            .form(&form_data)
+            .send()
+            .await
+            .map(|_| Ok(0))
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToDeletePackage))?
+    }
+
+    /// Add a user to an npm team
+    pub async fn add_user_to_team(&self, params: &str, module: &str) -> Result<i32, ApiError> {
+        self.login()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::LoginFlowError))?;
+        let params: AddRemoveUserToFromTeamParams =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+        let user = self.validate_npm_username(&params.user)?;
+        let team = self.validate_npm_team_name(&params.team)?;
+
+        info!(
+            "Adding user [{}] to team [{}] on behalf of [{module}]",
+            user, team
+        );
+
+        let csrf_token = self
+            .get_csrftoken_from_cookies()
+            .map_err(|_| ApiError::NpmError(NpmError::WrongClientStatus))?;
+        let body = format!(r#"{{"csrftoken": "{}", "user": "{}"}}"#, &csrf_token, user);
```

Comment:
This should really be it's own structure then serialized properly. The validators *should* prevent any issues here, but only if they stay compatible with this kind of interpolation. Safer to just use `serde_json` serialize

Reply:
Fixed!